src/net_processing.cpp

```diff
@@ -1142,60 +1143,71 @@ void static ProcessGetBlockData(CNode* pfrom, const Consensus::Params& consensus
         std::shared_ptr<const CBlock> pblock;
         if (a_recent_block && a_recent_block->GetHash() == pindex->GetBlockHash()) {
             pblock = a_recent_block;
+        } else if (inv.type == MSG_WITNESS_BLOCK) {
+            // Fast-path: in this case it is possible to serve the block directly from disk,
+            // as the network format matches the format on disk
+            LogPrintf("debug: Serving raw block directly from disk: %s\n", pindex->ToString());
```

Comment:
Should probably remove this debug logging

Reply:
Yes, definitely. I added it while WIP so that people testing this can be sure that the code actually triggers and they're testing the right thing.